src/Entity/Enemy.cpp

```diff
@@ -47,76 +67,187 @@ void InitEnemy(flecs::world& ecs)
 															5})
 							.emplace_override<AnimationComponent>(5, 0.1f, 0.0f, 0)
 							.emplace_override<AnimationStateComponent>(AnimationName::IDLE);
+							
 
 	ecs.system("SpawnNewEnemy")
 			.kind(flecs::OnUpdate)
 			.iter(SpawnNewEnemy);
 
-	ecs.system<TransformComponent, VelocityComponent>()
+	
+	ecs.system<TransformComponent, VelocityComponent, TargetTowerComponent>()
 			.with<EnemyTag>()
 			.kind(flecs::OnUpdate)
 			.each(EnemyUpdate);
 }
 
+//Spawn support functions
 void SpawnNewEnemy(flecs::iter& iter)
 {
+	// Static variables to keep track of the state
+	static float lastWaveEndTime = 0.0f;// Initialized to 0 to indicate that we haven't had a wave yet
 	static float lastSpawnTime = 0.0f;
-	float currentTime = GetTime();
+	static int cornerIndex = -1;// Will be set in the first wave
 
-	if (currentTime - lastSpawnTime < 1.0f)// Spawn a new enemy every 1 seconds
+	float currentTime = GetTime();
+	
+	// Check if it's time to start a new wave and spawn an enemy
+	if (!CheckAndStartNewWave(lastWaveEndTime, cornerIndex, currentTime))
 		return;
-
+	
+	//DEBUG
+	gCurrenSpawnDurationDebug = currentTime - lastSpawnTime;
+	// Spawn a new enemy every ENEMIES_SPAWN_INTERVAL
+	if (currentTime - lastSpawnTime < ENEMIES_SPAWN_INTERVAL)
+		return;
+	//Reset time only when spawn was succesfull?
 	lastSpawnTime = currentTime;
+	// Spawn the enemy
+	SpawnEnemy(iter.world(), cornerIndex, lastSpawnTime, currentTime);
 
-	const TransformComponent transformComponent{{0.0f, 0.0f}, gCharacterSize};
-	const Rectangle newGoblinRect{transformComponent.mPosition.x, transformComponent.mPosition.y, gCharacterSize.x, gCharacterSize.y};
+}
 
-	bool hasCollided{false};
+bool CheckAndStartNewWave(float& lastWaveEndTime, int& cornerIndex, float currentTime)
+{
+	// Determine the current state: wave or pause
+	bool inWave = (currentTime - lastWaveEndTime) < (WAVE_DURATION + PAUSE_DURATION) && (currentTime - lastWaveEndTime) > PAUSE_DURATION;
```

Comment:
This kind of condition can be difficult to read the more terms you add. To make it more readable, you can extract it in its own function, and have each condition on a line with a comment:
```cpp
bool IsWaveInProgress()
{
    // Check that the elapsed time since the end of the last wave…
    if ((currentTime - lastWaveEndTime) >= (WAVE_DURATION + PAUSE_DURATION))
        return false;
    […]
    return true;
}
```